daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala

```diff
@@ -269,6 +290,23 @@ class TransactionCoderSpec
       }
     }
 
+    "fail if try encode rollback node in version < minExceptions" in {
```

Comment:
Please also add a test for decoding

Reply:
So what are we trying to test here? .. _that we cannot decode a rollback node as a transaction with version < `minExceptions`_

But I'm not sure how this can be tested:

- We must create a proto with a rollback node
- and this must be created with a tx-version `>= minExceptions`, or else the encode will fail.

- Then we must attempt to decode that proto with a target version of `< minExceptions`.
- But this will fail with the general check in `decodeNodeVersion`, that: _A transaction of version V-OLD cannot contain node of newer version (v-NEWER)_

- So we never reach the code which fails if it find a rollback node in an old version.


By this logic, the version check for `ROLLBACK` in `decodeNode` is redundant.

However, I remember in previous similar situations, Remy is keen we keep such checks even if that cannot be hit.

Reply:
If we assume that our code is correct, we don’t need any of those checks since we will never produce those nodes for the wrong version. The point of those checks is to be extra safeguards so encoding doesn’t make assumption about the producer that passes in the un-encoded version and decoding doesn’t make assumptions about the encoded version.

For your example, can we encode it, then tweak the version in the encoded protobuf and then decode it and see that it fails?

Reply:
Makes sense.
I coded this by disabling the check at encode time, thus allowing the bad proto to be created.